### PR TITLE
Feature/57-Create-a-calendar-page

### PR DESCRIPTION
**Pull request by baja156**
- [ ] The pull request is complete according to the following criteria:
  - [x] Acceptance criteria have been met
  - [x] The documentation is kept up-to-date
  - [ ] Comprehensive tests (if applicable) have been generated and all pass.
  - [ ] The pull request describes the changes that have been made, and enough information is present in the description for any developer to understand what has changed
  - [ ] Commits have been squashed (or will be on merge).
  - [ ] The branch name is descriptive and follows the pull request title format : {issue/bug...}/(Issue Number) - Name of issue. E.g bug/30-Fix-Project
  - [ ] The pull request title is of the following format : {issue/bug...}/(Issue Number) - Name of issue. E.g bug/30-Fix-Project
  - [ ] The description uses [github syntax](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) to link to the issue. E,g Resolves se701g2/Doto#{Number}
  - [ ] At least two reviewers assigned. One of which must be the assigner of the issue.
  - [ ] If there are merge conflicts, run git rebase as opposed to git merge with master.

### CODE REVIEW

**Comment by daltonlim** [comment]
Hi, just wondering if you ran git merge as opposed to git rebase.

**Comment by baja156** [comment]
> Hi, just wondering if you ran git merge as opposed to git rebase.

I have been merging not rebasing

File: doto-frontend/src/components/pages/Calendar/Calendar.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        const tempAppointments = [
+            {
+                title: "Lecture-1",
+                descrition: "Lecture at Auckland Uni",
+                startDate: new Date(2020, 2, 12, 9, 0), // 0 is January
+                endDate: new Date(2020, 2, 12, 11, 0),
+                id: 0,
+                priority: 3,
+                location: "Room 1",
+                color: "blue",
```
Comment (lakshyaoberoi) [inline doto-frontend/src/components/pages/Calendar/Calendar.js]: Based on our add task properties, would we not require a duration field as well?

Comment (Dinith1) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: Oh yea, it should have duration. Although, couldn't this be inferred from `endDate - startDate`? is duration supposed to correspond to the dates, or is it separate? @lucycy98 pls help

Comment (baja156) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: Start date and end date is required by the calendar component we're using, sorry can't use duration. Backend should've configured it to be start date and end date, cos thats what I asked for

Comment (Dinith1) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: wdym? Is duration shown on the frontend somwhere?

Comment (baja156) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: duration as in the field at the modal? or duration as in the duration of the task you see on the calendar

Comment (Dinith1) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: > duration as in the field at the modal? or duration as in the duration of the task you see on the calendar

Both? I'm confused now sorry :/

Comment (baja156) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: In both cases I mentioned, duration is shown on the front end

Comment (lucycy98) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: Yea we don't have duration in back end, cause frontend wanted start and end date. Originally I thought we would just have a due date and duration so we can use scheduling to determine a time

Comment (baja156) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: Like @Dinith1 said, duration can be inferred from the start/end date though right?
> Yea we don't have duration in back end, cause frontend wanted start and end date. Originally I thought we would just have a due date and duration so we can use scheduling to determine a time

Comment (lakshyaoberoi) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: In the Modal, we do not have a field for Start Date though. So where in the front-end is this being used?

Comment (baja156) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: the calendar component (the react scheduler), needs start and end date to make an event. thats why i asked back end to explicitly store those 2, and these can be inferred from the duration when the user selects due date and duration

File: doto-frontend/src/components/pages/Calendar/Calendar.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        <div className="CalendarPage">
+            <div className="Side" />
+            <span className="container">
+                {/* Header is a placeholder */}
+                <header className="Calendar-header">
+                    {/* eslint-disable-next-line react/no-unescaped-entities */}
+                    <span>Calendar</span>
+                </header>
+                <div className="flex justify-end">
+                    <div className="ml-3">
```
Comment (Cvos0162) [inline doto-frontend/src/components/pages/Calendar/Calendar.js]: We might want to contain all div/modal with functionality into individual function/classes.
So it is much easier to handle plus its more self-documenting code

Comment (Cvos0162) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: so it  can be div on line 90, div on line 95, and the Modal on line 116

Comment (baja156) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: I like this too but I was thinking of merging this and then, sitting with mario and doing integration where we'd separate this functionality later on today. In a different PR

Comment (Cvos0162) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: Ok lets create a issue to resolve this. And reference it here.

Comment (baja156) [reply doto-frontend/src/components/pages/Calendar/Calendar.js]: @Cvos0162  On it mate

**Pull request merged**